src/plugins/formulas/__tests__/formulas.spec.js

```diff
@@ -1048,30 +1023,299 @@ describe('Formulas general', () => {
       expect(hot.getSourceDataAtRow(3)).toEqual([2011, 4, 2517, 4822, 552, 6127]);
-      expect(hot.getSourceDataAtRow(4)).toEqual([2012, '=Sum(a2:a5)', '=SUM(B5,E3)', '=A2/B2', 12, '\'=SUM(E5)']);
+      expect(hot.getSourceDataAtRow(4)).toEqual([2012, '=Sum(a2:a5)', '=SUM(B5,E3)', '=A2/B2', 12, '=SUM(E5)']);
 
       hot.redo();
 
       expect(hot.getSourceDataAtRow(0)).toEqual(['=$B$2', 'Maserati', 'Mazda', 'Mercedes', 'Mini', '=A$1']);
       expect(hot.getSourceDataAtRow(1)).toEqual([2009, 0, 2941, 4303, 354, 5814]);
       expect(hot.getSourceDataAtRow(2)).toEqual([2011, 4, 2517, 4822, 552, 6127]);
-      expect(hot.getSourceDataAtRow(3)).toEqual([2012, '=SUM(A2:A4)', '=SUM(B4,#REF!)', '=A2/B2', 12, '\'=SUM(E5)']);
+      expect(hot.getSourceDataAtRow(3)).toEqual([2012, '=SUM(A2:A4)', '=SUM(B4,#REF!)', '=A2/B2', 12, '=SUM(E5)']);
 
       hot.redo();
 
       expect(hot.getSourceDataAtRow(0)).toEqual(['=$B$2', 'Maserati', 'Mini', '=A$1']);
       expect(hot.getSourceDataAtRow(1)).toEqual([2009, 0, 354, 5814]);
       expect(hot.getSourceDataAtRow(2)).toEqual([2011, 4, 552, 6127]);
-      expect(hot.getSourceDataAtRow(3)).toEqual([2012, '=SUM(A2:A4)', 12, '\'=SUM(E5)']);
+      expect(hot.getSourceDataAtRow(3)).toEqual([2012, '=SUM(A2:A4)', 12, '=SUM(E5)']);
 
       hot.redo();
 
       expect(hot.getSourceDataAtRow(0)).toEqual([2011, 4, 552, 6127]);
-      expect(hot.getSourceDataAtRow(1)).toEqual([2012, '=SUM(A1:A2)', 12, '\'=SUM(E5)']);
+      expect(hot.getSourceDataAtRow(1)).toEqual([2012, '=SUM(A1:A2)', 12, '=SUM(E5)']);
+    });
+  });
+
+  describe('autofill', () => {
+    const fillHandleSelector = '.wtBorder.current.corner';
+
+    const autofill = (endRow, endCol) => {
+      spec().$container.find(fillHandleSelector).simulate('mousedown');
+
+      spec().$container
+        .find(`tbody tr:eq(${endRow}) td:eq(${endCol})`)
+        .simulate('mouseover')
+        .simulate('mouseup');
+    };
+
+    // Most of these tests will produce invalid values (out of bound addresses,
+    // #CYCLE! errors), but we only care about the formula offsets.
+    //
+    // https://docs.google.com/spreadsheets/d/1ERI3YEe7GYWUKdKGPU4C97yUh1fOM6HILZY03AB8wwk/edit?usp=sharing
+    it('single cell, down', () => {
+      const hot = handsontable({
+        data: [
+          ['=A1'],
+          ['x'],
+          ['x']
+        ],
+        formulas: true
+      });
+
+      selectCell(0, 0);
+      autofill(2, 0);
+
+      expect(hot.getSourceData()).toEqual([
+        ['=A1'],
+        ['=A2'],
+        ['=A3']
+      ]);
+    });
+
+    it('single cell, right', () => {
+      const hot = handsontable({
+        data: [
+          ['=A1', 'x', 'x']
+        ],
+        formulas: true
+      });
+
+      selectCell(0, 0);
+      autofill(0, 2);
+
+      expect(hot.getSourceData()).toEqual([
+        ['=A1', '=B1', '=C1']
+      ]);
+    });
+
+    it('range, down, partial', () => {
```

Comment:
These descriptions could be more descriptive (and in a form of `it('should do something when something happens')`.

Goes for all the `it`s in this section - for example, I don't really know what `it('range, up, even')` means.

Reply:
I second that ☝️

Reply:
I tried https://github.com/handsontable/handsontable/pull/7648/commits/1289090733b38b6c8451761f64961ae899cce02b